storage/merkle/tree.go

```diff
@@ -13,13 +13,10 @@ import (
 // maxKeyLength in bytes:
 // For any key, we need to ensure that the entire path can be stored in a short node.
 // A short node stores the _number of bits_ for the path segment it represents in 2 bytes.
-// However, a short node with zero path length is not part of our storage model. Therefore,
-// we use the convention:
-//  * for path length l with 1 ≤ l ≤ 65535: we represent l as unsigned int with big-endian encoding
-//  * for l = 65536: we represent l as binary 00000000 00000000
-// This convention organically utilizes the natural occurring overflow and is therefore extremely
-// efficient. In summary, we are able to represent key length of up to 65536 bits, i.e. 8192 bytes.
-const maxKeyLength = 8192
+// Note that a short node with zero path length is not part of our storage model.
+// In summary, we are able to represent key lengths multiple of 8 of up to 65535 bits, i.e. less than 8191 bytes.
+const maxKeyLenBits = 65535
+const maxKeyLength = maxKeyLenBits / 8
```

Comment:
To me, this is confusing because we state essentially conflicting information about the max possible key length:
* we say that a key can at most be `8191` _bytes_ long (which is `65528` bits)
* but we also state that a key can be `65535` bits long

My suggestion would be to just use 

```suggestion
// Hence, the theoretically possible value range is [0,65535]. However, a short node with
// zero path length is not part of our storage model. Furthermore, we always represent
// keys as _byte_ slices, i.e. their number of bits must be an integer-multiple of 8.
// Therefore, the longest possible key is 8191 bytes long. In summary, the range of valid
// values for a short node's length in bits is [1, 65528].
const maxKeyLength = 8191
```

and adjust the [test](https://github.com/onflow/flow-go/blob/8f41c5c459f4be76e3afb64f4b11bbfd5b2d905d/storage/merkle/node_test.go#L87-L104) to use a `8191`-byte key

Reply:
I think there is still an advantage of keeping `maxKeyLenBits`. 
The sentence says that we support "key length multiple of 8 bits, up to 65535". Since 65535 in not a multiple of 8, the largest possible length is the multiple of 8 right before, which is 65528. 

This is clearly confusing so I'll try to rephrase it and remove mentions of 65535.

Reply:
[a21444e](https://github.com/onflow/flow-go/pull/1909/commits/a21444e0faada936ce5994b66642e5fa7c8ad414)